Re-add djbdns, ucspi-tcp and daemontools

dermoth opened the pull request:
Maintainer: Thomas Guyot-Sionnest &lt;tguyot@gmail.com&gt; (dermoth)
Compile tested: arm_cortex-a9_vfpv3 (Lede reboot-4553-ga89c36b508)
Run tested: Lede Reboot 17.01.2 r3435-65eec8bd5f, mvebu (WRT3200ACM), tested dnscache running from daemontools)

Description:

I'm upgrading my router from an older OpenWRT build, using Lede Reboot (17.01.2 r3435-65eec8bd5f). My setup was thighly integrated with dnscache so I re-added djbdns and its dependencies to the packages stream so I could build/install them.

zorun commented:
Hi, thanks for the work!

Just a few comments:

- you could switch the init scripts to procd, see https://wiki.openwrt.org/inbox/procd-init-scripts
- is the `ignoreip-update` script really needed?
- you carry a lot of patches for djbdns, none of them made it upstream?
- please sign-off all your commits
- instead of extending the OpenWRT coypright, keep it as it is and add your own
- I don't think bumping `PKG_RELEASE` is needed, you can just reset it to 1 or leave it as it was in oldpackages

Maybe other people also have comments (it's a large pull request).

dermoth commented:
@zorun Thanks for the feedback

* you could switch the init scripts to procd, see https://wiki.openwrt.org/inbox/procd-init-scripts
  * I'll have a look; I don't personally use init scripts, I use daemontools instead (adding a line to inittab to run svscanboot). I haven't modified or tested this part...
* is the ignoreip-update script really needed?
  * It's optional, but since it was there I decided to use it myself and enhanced it a little bit
* you carry a lot of patches for djbdns, none of them made it upstream?
  * Djbdns was developed by Daniel J. Bernstein over a decade ago (with no critical security vulnerabilities to date), then years later released in the public domain. The standard practice before it became public was to distribute patches to add features and fixes. The alternative would be to find a new tree that is maintained and use it/submit any bug upstream. Even Debian still use the original package + patches.
* please sign-off all your commits
  * Will do. I'm new; this came up in another PR
* instead of extending the OpenWRT coypright, keep it as it is and add your own
  * Is there any rationale? I don't care about Copyright, I might as well leave the lines untouched if you prefer. IMHO it's always easier to manage legal issues when there's less copyright holders...
* I don't think bumping PKG_RELEASE is needed, you can just reset it to 1 or leave it as it was in oldpackages
  * Ok I'll reset to 1... Looks like installed packages aren't carried over from previous version anyway.

zorun commented:
For the copyright, better not to touch it then.  The rationale is that you are not under contract with OpenWrt so it does not make sense to extend the copyright like this.  I'm not sure it makes any difference, but well.

champtar commented:
Travis tests

dermoth commented:
I corrected the requirements for commit message formats (first commit split in 3 for that reason), and also added a new patch from an old OpenWRT tracker item that never got merged... it fixes a transient cname issue I managed to reproduce recently.

Also fixed minor build issues (using PKG_HASH instead of PKG_MD5SUM) and reset the djbdns release number.

dermoth commented:
Strange, I rebased the pr after testing on latest release... 18.06.1-mvebu-cortexa9, and it implicitly closed it.
 Can it be re-opened? I will open a new pr and reference this one anyway...

dermoth commented:
Oh got it, I accidentally pushed upstream's master into this one when I switch to another clone... Trying to reopen now.
